lib/core/src/Cardano/Wallet/Api/Types.hs

```diff
@@ -402,6 +408,19 @@ newtype ApiSelectCoinsData (n :: NetworkDiscriminant) = ApiSelectCoinsData
     { payments :: NonEmpty (AddressAmount (ApiT Address, Proxy n))
     } deriving (Eq, Generic, Show)
 
+data ApiSelectCoinsActionData (n :: NetworkDiscriminant)
+    = ApiSelectCoinsActionData
+    { action :: W.DelegationAction
+    , passphrase :: ApiT (Passphrase "raw")
+    }
+    deriving (Eq, Generic, Show)
+
+data ApiCoinSelectionCert (n :: NetworkDiscriminant) = ApiCoinSelectionCert
+    { inputs :: !(NonEmpty (ApiCoinSelectionInput n))
+    , outputs :: !(NonEmpty (AddressAmount (ApiT Address, Proxy n)))
+    , certs :: ! [ApiT Certificate]
+    } deriving (Eq, Generic, Show)
```

Comment:
What about simply extending the current `ApiCoinSelection` data-type? The list of certificates can be empty for other delegation and we need not to define yet-another-resource.

Reply:
Isn't the point of types to have clear separation? For the old endpoint, `certs` will always be null. That should be expressed in the type system, so a potential client knows it doesn't have to deal with it.

Reply:
> For the old endpoint,

That's the point though, it is the _same_ endpoint, manipulating the same resource. Just a different request payload. It would indeed be always empty for normal payments, and that is just right because normal payments don't have any certificates. The introduction of certificates done by this PR is an extension of the existing resource, not a new definition.